app/routes/_gcn.circulars/circulars.server.ts

```diff
@@ -173,12 +201,31 @@ export async function search({
 }
 
 /** Get a circular by ID. */
-export async function get(circularId: number): Promise<Circular> {
+export async function get(
+  circularId: number,
+  version?: number
+): Promise<Circular> {
   if (isNaN(circularId)) throw new Response(null, { status: 404 })
   const db = await tables()
-  const result = await db.circulars.get({
-    circularId,
-  })
+
+  let result = version
+    ? await db.circulars_history.get({
+        circularId,
+        version,
+      })
+    : await db.circulars.get({
+        circularId,
+      })
+
+  if (!result && version && !isNaN(version)) {
```

Comment:
Remember that `Boolean(NaN) === false`. This `if` statement is not correct.

Reply:
ah good catch, I think I am misunderstanding what this case should be catching. Is this correct:

If the result isn't found in the initial check given a `circularId` and `version` (which is a number), just get whatever the latest/current version is. 

Then in the following block, if the provided version doesn't match on the found result, clear the result

Reply:
Correct.

The following possible values of type `number | undefined` are falsey:

- undefined: no version passed as an argument, so don't even bother looking up by version
- NaN: an impossible value, should always throw 404
- 0: another impossible value, should always throw 404

Reply:
Gotcha, I had made the change to that if statement because of this error:
![image](https://github.com/nasa-gcn/gcn.nasa.gov/assets/11023698/f3296038-879e-4a47-a6b6-1640b4a65fb2)

Reply:
To satisfy type inference, you chould check that `number !== undefined && !number`.